src/services/dot.ts

```diff
@@ -1,309 +1,21 @@
-   * @param memberAccount
-   * @param amountDot
-   */
-  async craftUnbondFromPoolTx(memberAccount: string, amountDot: number): Promise<DotTx> {
-    const amountPlanck = this.testnet ? this.wndToPlanck(amountDot.toString()) : this.dotToPlanck(amountDot.toString());
-    const { data } = await api.post<DotTx>(`/v1/dot/transaction/unbond-pool`, {
-      member_account: memberAccount,
-      amount_planck: amountPlanck,
-    });
-    return data;
-  }
-
-  /**
-   * Craft a pool withdraw unbonded transaction
-   * Withdraw unbonded funds from member_account.
-   * @param memberAccount
-   */
-  async craftWithdrawUnbondedFromPoolTx(memberAccount: string): Promise<DotTx> {
-    const { data } = await api.post<DotTx>(`/v1/dot/transaction/withdraw-unbonded-pool`, {
-      member_account: memberAccount,
-    });
-    return data;
-  }
-
-  /**
-   * Sign transaction with given integration
-   * @param integration custody solution to sign with
-   * @param tx raw transaction
-   * @param note note to identify the transaction in your custody solution
-   */
-  async sign(integration: Integration, tx: DotTx, note?: string): Promise<DotSignedTx> {
-    const payload = {
-      rawMessageData: {
-        messages: [
-          {
-            content: tx.data.unsigned_tx_payload.substring(2),
-          },
-        ],
-      },
-    };
-
-    const fbSigner = this.getFbSigner(integration);
-    const fbNote = note ? note : "DOT tx from @kilnfi/sdk";
-    const fbTx = await fbSigner.sign(payload, this.testnet ? "WND" : "DOT", fbNote);
-    const signature = `0x00${fbTx.signedMessages![0].signature.fullSig}`;
-
-    const { data } = await api.post<DotSignedTx>(`/v1/dot/transaction/prepare`, {
-      unsigned_tx_serialized: tx.data.unsigned_tx_serialized,
-      signature: signature,
-    });
-    data.data.fireblocks_tx = fbTx;
-    return data;
-  }
-
-  /**
-   * Broadcast signed transaction
-   * @param signedTx
-   */
-  async broadcast(signedTx: DotSignedTx): Promise<DotTxHash> {
-    const { data } = await api.post<DotTxHash>(`/v1/dot/transaction/broadcast`, {
-      tx_serialized: signedTx.data.signed_tx_serialized,
-    });
-    return data;
-  }
-
-  /**
-   * Get transaction status
-   * @param txHash transaction hash
-   */
-  async getTxStatus(txHash: string): Promise<DotTxStatus> {
-    const { data } = await api.get<DotTxStatus>(`/v1/dot/transaction/status?tx_hash=${txHash}`);
-    return data;
+export class DotService extends SubstrateService {
+  constructor(props: ServiceProps) {
+    super(props, "DOT");
   }
 
   /**
-   * Decode transaction
-   * @param txSerialized transaction serialized
+   * Convert DOT token to planck
+   * @param amount amount in DOT
    */
-  async decodeTx(txSerialized: string): Promise<UnsignedTransaction> {
-    const { data } = await api.get<UnsignedTransaction>(`/v1/dot/transaction/decode?tx_serialized=${txSerialized}`);
-    return data;
+  mainToPlanck(amount: string): string {
```

Comment:
why name it "generically" if it does not do the same for KSM and DOT ?

Reply:
this function has the same role for all substrate tokens, converting from the main token to planck however the conversion rate is different for DOT, WND & KSM hence the `abstract` keyword. Does this not look straight forward to you?

Reply:
for all other protocols we have a naming with token names no ?

Reply:
for consistency I guess we should have ksmToPlanck and dotToPlanck no ?

Reply:
Ok let me find a workaround. Let me know if this looks good to you